extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisSkippedRepositoriesTab.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { VariantAnalysisSkippedRepositoryGroup } from '../../remote-queries/shared/variant-analysis';
+import { Alert } from '../common';
+import { VariantAnalysisSkippedRepositoryRow } from './VariantAnalysisSkippedRepositoryRow';
+
+export type SkippedRepositoriesReason = 'no_access' | 'no_database';
```

Comment:
An alternative for this pattern would be to pass the alert title and message into the component. This would decouple the alert from the display of the rows and make it easier to re-use this component.

This would change it from:

```tsx
<VariantAnalysisSkippedRepositoriesTab
  reason='no_access'
  skippedRepositoryGroup={notFoundRepos} />
```

to:

```tsx
<VariantAnalysisSkippedRepositoriesTab
  alertTitle="No access"
  alertMessage="The following repositories could not be scanned because you do not have read access."
  skippedRepositoryGroup={notFoundRepos} />
```

That would remove quite a few lines from this component which are now only used to reduce the number of props that's passed into the component.

Reply:
Thanks that's a really good suggestion. Definitely helps to reduce the complexity of that component and the number of methods taking lots of arguments.